"Save File" and "Save File as"

TheMathix opened the pull request:
I changed the "Save File" functionality to subscribe automatically the file that was opened. At the same time, I added a "Save File as" function that does exactly what the previous function did. Implemented a dialog box asking if user wants to 'Save File' before opening another file. Only asks this if it's not the first file opened.

TEhhhA commented on js/render_logic/electron_inter.js:
```diff
@@ -10,10 +10,30 @@ const Armv5 = require('./js/render_logic/armv5/armv5.js');
+                if (err) throw err;
+            });
+
+        } else {
```
I don't see the point of this empty else statement.

TEhhhA commented on js/render_logic/electron_inter.js:
```diff
@@ -10,10 +10,30 @@ const Armv5 = require('./js/render_logic/armv5/armv5.js');
 
 //Listen to event "Open file" from main.js
 ipcRenderer.on('Open file', (event, arg) => {
+	
```
A nitpick, I know the style is not hyper defined but I would prefer if it stayed somewhat consistent. Please don't add stray \n on function definition (between { and first line of function).

TheTarados commented:
Hi, thanks for your contribution. I checked a little bit for style, I will try the functionality today or tomorrow.

TheTarados commented:
Also, if you have the possibility to edit the commit message, I would appreciate to have a clean and short one liner followed by the more complete explanation you have given.

TEhhhA commented on js/render_logic/electron_inter.js:
```diff
@@ -91,6 +111,13 @@ ipcRenderer.on('Close file', (event, arg) => {
 })
 
 ipcRenderer.on('Save file', (event, arg) => {
+
```
no \n

TheTarados commented:
You should also modify the readme to reflect changes

TheMathix committed:
Save File and Save File As

Changed the "Save File" functionality to subscribe automatically the file that was opened. At the same time, it was added a "Save File as" function that does exactly what the previous function did. Implemented a dialog box asking if user wants to 'Save File' before opening another file. Only asks this if it's not the first file opened.

TheTarados commented:
The code is good now but you should squash the two commits to have a single one.
At the same time, rename the one-liner to  "Save File only subscribes the file. Add Save File As submenu.", it is still too long at the moment.

TheMathix commented:
I hope I got it right this time, sorry for my inexperience. Thanks for the patience.

Best regards,
Matheus Ramos
________________________________
De: Thomas ANTOINE ***@***.***>
Enviado: terça-feira, 21 de janeiro de 2025 06:04
Para: TheTarados/AssDeb ***@***.***>
Cc: Themathix ***@***.***>; Author ***@***.***>
Assunto: Re: [TheTarados/AssDeb] "Save File" and "Save File as" (PR #4)


The code is good now but you should squash the two commits to have a single one.
At the same time, rename the one-liner to "Save File only subscribes the file. Add Save File As submenu.", it is still too long at the moment.

—
Reply to this email directly, view it on GitHub<https://github.com/TheTarados/AssDeb/pull/4#issuecomment-2603906433>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALKENJJ7VDFGYO55HR2IFJT2LX5QRAVCNFSM6AAAAABUSOOS26VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMBTHEYDMNBTGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

TheTarados commented:
Seems good to me :) Thank you for your contribution
